uuid.go

```diff
@@ -344,6 +345,30 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+// MarshalJSON marshalls the NullUUID as nil or the nested UUID
+func (u NullUUID) MarshalJSON() ([]byte, error) {
+	if u.Valid == false {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(u.UUID)
+}
+
+// UnmarshalJSON unmarshalls a NullUUID
+func (u *NullUUID) UnmarshalJSON(b []byte) error {
+	var s *UUID
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	if s != nil {
+		u.Valid = true
+		u.UUID = *s
+	} else {
+		u.Valid = false
+	}
+	return nil
+}
```

Comment:
I highly prefer being explicit about return values, I think it's much more readable to see `return err` or `return nil` than just seeing `return` and trying to infer what's being returned.

According to [Go style](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters):

> Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values.

I'd say this is a medium sized function.